qiskit_experiments/framework/experiment_data.py

```diff
@@ -243,9 +244,12 @@ def __init__(
         self._backend = None
         if backend is not None:
             self._set_backend(backend, recursive=False)
+        self._provider = provider
+        if provider is None and backend is not None:
+            self.provider = backend.provider
         self._service = service
-        if self._service is None and self.backend is not None:
-            self._service = self.get_service_from_backend(self.backend)
+        if self._service is None and self.provider is not None:
+            self._service = self.get_service_from_provider(self.provider)
```

Comment:
Usually when running an experiment, the user has a concrete backend used for that experiment, and it passes to the experiment data. However, when loading an experiment data via `ExperimentData.load()` one usually does not have any concrete backend. Instead, the user can either use the experiment service itself. Since most users are familiar with the provider but not with the experiment service, it's better to allow them to pass the provider itself. Indeed this should be documented.